File: 217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();

        unordered_map<int,int> mp;


        for(int num : nums){
            mp[num]++;
        }


        for(auto &i : mp){
            if(i.second >=2){
                return true;
            }
        }

        return false;
    }
};

File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        

        HashMap<Integer,Integer> map = new HashMap<>();


      for(int n : nums){
        map.put(n,map.getOrDefault(n,0)+1);
      }

      for(Map.Entry<Integer,Integer> entry : map.entrySet()){
          if(entry.getValue()>=2){
            return true;
          }
      }
      return false;
    }
}